File: backend/DTO/Response/ProductResponse.cs
using System.ComponentModel.DataAnnotations;

namespace backend.DTO.Response
{
    public class ProductResponse
    {
        [Required]
        public required int ProductId { get; set; }
        [Required]
        public required string Model { get; set; }
        [Required]
        public required string ProductLink { get; set; }
        [Required]
        public string? Description { get; set; }
        [Required]
        public required decimal Price { get; set; }
        [Required]
        public required int Stock { get; set; }
        [Required]
        public required int CategoryId { get; set; }
    }
}

File: frontend_admin/lib/model/response/product_response.dart
import 'package:equatable/equatable.dart';

class ProductResponse extends Equatable {
  const ProductResponse({
    this.productId,
    this.model,
    this.description,
    this.price,
    this.stock,
    this.categoryId,
    this.productLink,
  });

  factory ProductResponse.fromJson(Map<String, dynamic> json) {
    return ProductResponse(
      productId: json['productId'] as int?,
      model: json['model'] as String?,
      description: json['description'] as String?,
      price: json['price'] as double?,
      stock: json['stock'] as int?,
      categoryId: json['categoryId'] as int?,
      productLink: json['productLink'] as String?,
    );
  }
  final int? productId;
  final String? model;
  final String? description;
  final double? price;
  final int? stock;
  final int? categoryId;
  final String? productLink;

  Map<String, dynamic> toJson() {
    return {
      'productId': productId,
      'model': model,
      'description': description,
      'price': price,
      'stock': stock,
      'categoryId': categoryId,
      'productLink': productLink,
    };
  }

  @override
  List<Object?> get props => [
    productId,
    model,
    description,
    price,
    stock,
    categoryId,
    productLink,
  ];
}
